mongodbatlas/data_source_mongodbatlas_cloud_shared_tier_restore_jobs.go

```diff
@@ -0,0 +1,135 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"job_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_deployment_item_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_scheduled_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"delivery_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expiration_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasCloudSharedTierRestoreJobRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).AtlasV2
+
+	projectID := d.Get("project_id").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	jobs, resp, err := conn.SharedTierRestoreJobsApi.ListSharedClusterBackupRestoreJobs(ctx, clusterName, projectID).Execute()
+	defer resp.Body.Close()
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting shared tier restore jobs for cluster '%s': %w", clusterName, err))
+	}
+
+	if err := d.Set("results", flattenShardTierRestoreJobs(jobs.Results)); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `results`: %w", err))
+	}
+
+	if err := d.Set("total_count", jobs.TotalCount); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `total_count`: %w", err))
+	}
+
+	d.SetId(id.UniqueId())
+	return nil
+}
+
+func flattenShardTierRestoreJobs(sharedTierJobs []atlasSDK.TenantRestore) []map[string]interface{} {
+	if len(sharedTierJobs) == 0 {
+		return nil
+	}
+
+	results := make([]map[string]interface{}, len(sharedTierJobs))
+	for i := range sharedTierJobs {
+		sharedTierJob := &sharedTierJobs[i]
+		results[i] = map[string]interface{}{
+			"job_id":                      sharedTierJob.Id,
+			"status":                      sharedTierJob.Status,
+			"target_project_id":           sharedTierJob.TargetProjectId,
+			"target_deployment_item_name": sharedTierJob.TargetDeploymentItemName,
+			"snapshot_url":                sharedTierJob.SnapshotUrl,
+			"snapshot_id":                 sharedTierJob.SnapshotId,
+			"snapshot_finished_date":      sharedTierJob.SnapshotFinishedDate.String(),
```

Comment:
[Question] Do you think that sdk should have dates as strings?

If that would be the case then all your mappings would look like this. 
CLI also does that transformation. 

```
b, _ := json.Marshal(&u1)
	var m map[string]interface{}
	_ = json.Unmarshal(b, &m)
```